index-providers/modeshape-elasticsearch-index-provider/src/main/java/org/modeshape/jcr/index/elasticsearch/Operations.java

```diff
@@ -0,0 +1,351 @@
+            Comparison comp = (Comparison)constraint;
+            String field = (String) operand(comp.getOperand1()).apply(comp.getOperand1(), variables);
+            Object value = operand(comp.getOperand2()).apply(field, comp.getOperand2(), variables);
+            switch (comp.operator()) {
+                case EQUAL_TO:
+                    return QueryBuilders.matchQuery(field, value);
+                case GREATER_THAN:
+                    return QueryBuilders.rangeQuery(field).gt(value);
+                case GREATER_THAN_OR_EQUAL_TO:
+                    return QueryBuilders.rangeQuery(field).gte(value);
+                case LESS_THAN:
+                    return QueryBuilders.rangeQuery(field).lt(value);
+                case LESS_THAN_OR_EQUAL_TO:
+                    return QueryBuilders.rangeQuery(field).lte(value);
+                case NOT_EQUAL_TO:
+                    return QueryBuilders.boolQuery().mustNot(
+                            QueryBuilders.rangeQuery(field).from(value).to(value));
+                case LIKE:
+                    String queryString = (String)value;                    
+                    
+                    if (!queryString.contains("%")) {
+                        return QueryBuilders.queryStringQuery(queryString);
+                    }
+                    
+                    String[] terms = queryString.split(" ");
+                    QueryBuilder[] termFilters = new QueryBuilder[terms.length];
+                    
+                    for (int i = 0; i < termFilters.length; i++) {
+                        termFilters[i] = terms[i].contains("%") ?
+                                QueryBuilders.wildcardQuery(field, terms[i].replaceAll("%", "*")) :
+                                QueryBuilders.queryStringQuery(terms[i]);
+                    }
+                    
+                    if (termFilters.length == 1) {
+                        return termFilters[0];
+                    }
+                    
+                    QueryBuilder builder = 
+                            QueryBuilders.boolQuery().must(termFilters[0]).must(termFilters[1]);
+                    for (int i = 2; i < termFilters.length; i++) {
+                        builder = QueryBuilders.boolQuery().must(builder).must(termFilters[i]);
+                    }
+                    
+                    return builder;
+            }
+        }
+
+        if (constraint instanceof SetCriteria) {
+            SetCriteria setCriteria = (SetCriteria)constraint;
+            
+            String field = (String) operand(setCriteria.getOperand()).apply(setCriteria.getOperand(), variables);
+            ArrayList list = new ArrayList();
+            for (StaticOperand so : setCriteria.getValues()) {
+                Object vals = operand(so).apply(field, so, variables);
+                if (vals instanceof Object[]) {
+                    list.addAll(Arrays.asList((Object[]) vals));
```

Comment:
The values will be of the type used within the query system. Are you sure these values don't need to be converted before they're passed into the query?